vortex-array/src/compute/take.rs

```diff
@@ -103,6 +125,46 @@ pub fn take(array: impl AsRef<Array>, indices: impl AsRef<Array>) -> VortexResul
     Ok(taken)
 }
 
+pub fn take_into(
+    array: impl AsRef<Array>,
+    indices: impl AsRef<Array>,
+    builder: &mut dyn ArrayBuilder,
+) -> VortexResult<()> {
+    let array = array.as_ref();
+    let indices = indices.as_ref();
+
+    debug_assert_eq!(
+        array.dtype(),
+        builder.dtype(),
+        "Take dtype mismatch {}",
+        array.encoding()
+    );
+
+    if !indices.dtype().is_int() {
+        vortex_bail!(
+            "Take indices must be an integer type, got {}",
+            indices.dtype()
+        );
+    }
+
+    let before_len = builder.len();
+
+    // We know that constant array don't need stats propagation, so we can avoid the overhead of
+    // computing derived stats and merging them in.
+    take_into_impl(array, indices, builder)?;
+
+    let after_len = builder.len();
+
+    debug_assert_eq!(
```

Comment:
There's some more asserts in the take recently since we cleaned up handling of nullable indices